cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -126,15 +172,238 @@ fn check_builder_metadata(_builder: impl ManifestBuilder) -> Result<CheckedBuild
     todo!()
 }
 
+fn gather_build_repo_meta(environment: &PathEnvironment) -> Result<LockedUrlInfo, PublishError> {
+    // Gather build repo info
+    let git = match environment.parent_path() {
+        Ok(env_path) => match GitCommandProvider::discover(env_path) {
+            Ok(git) => git,
+            Err(e) => return Err(PublishError::UnsupportEnvironmentState { msg: e.to_string() }),
+        },
+        Err(e) => return Err(PublishError::UnsupportEnvironmentState { msg: e.to_string() }),
+    };
+
+    let origin = match git.get_origin() {
+        Ok(origin) => origin,
+        Err(e) => return Err(PublishError::UnsupportEnvironmentState { msg: e.to_string() }),
+    };
```

Comment:
suggeston (style): Unless there is more involved handling of either case, `Error::map_err` can be be used as a shorthand to wrap errors


```suggestion
    let origin = git
        .get_origin()
        .map_err(|e| PublishError::UnsupportEnvironmentState { msg: e.to_string() })?;

```